### DISCUSSION TITLE
Negative -1 mass for all joints in the model.

### DISCUSSION THREAD

**Discussion by igor-sadalski** [stack-of-tasks/pinocchio#1711] [Q&A]
I am using URDF provided by pinocchio (one with defined mass and inertia values) my program is able to parse the URDF (e.g. I can change the name in the URDF and print it with model.name) but I am getting a fixed -1 mass for all joints. How to fix this?
```
const std::string urdf_filename = "/pinocchio/models/example-robot-data/robots/anymal_b_simple_description/robots/anymal.urdf";
pinocchio::Model model;
pinocchio::urdf::buildModel(urdf_filename,model);
pinocchio::Data data(model);

//all joints, including the mass[0] are equal to -1
for (int i = 0; i < model.nv; i++)
    {
        std::cout << data.mass[i] << std::endl;
    }

**Comment by jmirabel** [comment]
`data.mass` is the result of `computeSubtreeMasses` and a by product of some other algos. For the total mass only, use `computeTotalMass`.

In general, anything in `data` is computed by an algorithm. Here, the -1 is just a default value.

**Reply by jcarpent** [reply]
It is indeed a default value, made negavite, to popup some warnings in the user's mind.

**Reply by mehdi-benallegue** [reply]
Sorry, just passing by I have a question : When the value is not initializes shouldn't it throw an exception instead of a physically impossible value? (Or send a NaN? hope not)

I think it can avoid hidden bugs. I am not sure how it can happen but if this default value is added to other masses, the total would be positive and no bell would ring, especially if it does not make simulations diverge.

We had a somewhat similar case where a bug in the code allowing  to build the model of our robot in OpenHRP was giving the wrong sign to the non diagonal values of the inertia tensor. It took our team about 20 years to discover it when somebody happened to have a non SDP inertia matrix because of this bug.

**Reply by jcarpent** [reply]
We could too indeed. Thanks for the suggestion.
